File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/builder/vechicle/Plane.java
package com.bigboxcode.designpattern.builder.vechicle;

public class Plane {
    private int wheel;
    private int engine;
    private int seat;
    private int door;
    private int wing;

    private boolean interior;

    public Plane(int noOfWheel, int noOfEngine, int noOfSeat, int noOfDoor, int wing, boolean interior) {
        this.wheel = noOfWheel;
        this.engine = noOfEngine;
        this.seat = noOfSeat;
        this.door = noOfDoor;
        this.wing = wing;
        this.interior = interior;
    }

    public int getWheel() {
        return wheel;
    }

    public int getEngine() {
        return engine;
    }

    public int getSeat() {
        return seat;
    }

    public int getDoor() {
        return door;
    }

    public int getWing() {
        return wing;
    }

    public boolean isInterior() {
        return interior;
    }

    public String toString() {
        return "Plane: Wheel -> " + wheel + " | Engine -> " + engine + " | Seat -> " + seat + " | Door -> " + door + " | Wing: " + wing + " | Interior -> " + interior;
    }
}

File: design-patterns/php/builder/vehicle/Plane.php
<?php
// Plane.php

namespace Builder\Vehicle;


class Plane {
    public function __construct(
        private int $wheel,
        private int $engine,
        private int $seat,
        private int $door,
        private int $wing,
        private bool $interior) {
    }

    public function getWheel(): int {
        return $this->wheel;
    }

    public function getEngine(): int {
        return $this->engine;
    }

    public function getSeat(): int {
        return $this->seat;
    }

    public function getDoor(): int {
        return $this->door;
    }

    public function getWing(): int {
        return $this->wing;
    }

    public function isInterior(): bool {
        return $this->interior;
    }

    public function __toString(): string {
        return "Plane: Wheel -> " . $this->wheel . " | Engine -> " . $this->engine . " | Seat -> " . $this->seat . " | Door -> " . $this->door . " | Wing: " . $this->wing . " | Interior -> " . ($this->interior ? 'true' : 'false');
    }
}